File: Leetcode Daily Challenge/027-September-2025/012-vowels-game-in-a-string.java
// https://leetcode.com/problems/vowels-game-in-a-string/




class Solution {
    // TC : O(N)
    // SC : O(1)
    public boolean doesAliceWin(String s) {
        int cntVowels = 0;
        for (char ch : s.toCharArray()) {
            if (isVowel(ch)) cntVowels += 1;
        }
        return cntVowels == 0 ? false : true;
    }
    private static boolean isVowel(char ch) {
        return (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u');
    }
}

File: Leetcode Daily Challenge/027-September-2025/012-vowels-game-in-a-string.cpp
// https://leetcode.com/problems/vowels-game-in-a-string/




#include <bits/stdc++.h>
using namespace std;

class Solution {
private:
    bool isVowel(char &ch) {
        return (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u');
    }
public:
    // TC : O(N)
    // SC : O(1)
    bool doesAliceWin(string s) {   
        int vowelsCnt = 0;
        for (char &ch : s) {
            if (isVowel(ch)) vowelsCnt += 1;
        }
        return vowelsCnt == 0 ? false : true;
    }
};